OrbitClientData/ModuleManager.cpp

```diff
@@ -7,22 +7,25 @@
 #include <vector>
 
 #include "OrbitBase/Logging.h"
+#include "OrbitClientData/ModuleData.h"
 #include "absl/synchronization/mutex.h"
 #include "capture_data.pb.h"
 
 using orbit_client_protos::FunctionInfo;
 
 namespace OrbitClientData {
 
-void ModuleManager::AddNewModules(const std::vector<orbit_grpc_protos::ModuleInfo>& module_infos) {
+void ModuleManager::AddOrUpdateModules(
+    const std::vector<orbit_grpc_protos::ModuleInfo>& module_infos) {
   absl::MutexLock lock(&mutex_);
 
   for (const auto& module_info : module_infos) {
     auto module_it = module_map_.find(module_info.file_path());
     if (module_it == module_map_.end()) {
-      const auto [inserted_it, success] =
-          module_map_.try_emplace(module_info.file_path(), module_info);
+      const auto [_, success] = module_map_.try_emplace(module_info.file_path(), module_info);
```

Comment:
While in other languages `_` is part of the language or the standard for unused variables, I'm really not sold on this in C++. Why not just
```
const bool success = module_map_.try_emplace(module_info.file_path(), module_info).second;
```
?

Reply:
Not sure if I'm missing something here, but shouldn't this trigger an unused variable error?

(On the use of _: I don't feel strongly either way. I think it's fine if as a team we agree on this convention. In this particular case the use of .second works just as well for me, though.)

Reply:
In C++ there is a series of rules that determine if an identifier starting with an underscore is reserved and shouldn't be used. This specific name for a local variable doesn't fall into one of the reserved category, but still, in light of this, the common approach is to just avoid identifiers that start with underscores (including this degenerate case), to avoid doubts. So I'm not agreeing on this convention, as convenient as it is in other languages.

Reply:
I started using this underscore syntax a while ago. I wasn't sure about it, because the style guide does not mention it, hence I started a thread in chat about it and the conclusion was that it is fine when used in a local scope. Also, the compiler optimizes it away anyways. 
I use it mainly in for loops with hash maps and I think it improves readability
```
for (const auto& [_, value] : hash_map) { ... } 
```
instead of 
```
for (const auto& pair: hash_map) {
  const auto& value = pair.second;
  ...
}
```

That said, in this case I actually like your suggestion Davide and changed it.

Reply:
But then why not use an actual name instead of `_`, even if you're not using it? What's the problem with `for (const auto& [key, value] : hash_map) { ... }` instead of `for (const auto& [_, value] : hash_map) { ... } `?
`_` doesn't have a special meaning like in other languages, so you're still having an actual variable, which is now also weirdly named. This also means that if you do the same below or in a nested scope you actually have a redefinition, which goes directly against the purpose of this (which would be not declaring a variable at all).

Reply:
We discussed this before and agreed that it's fine. Unless there are strong (newly known) reasons (which I don't think there are), I'm fine if folks continue using it.

Reply:
I found another resource which has an opinion about it. Taking this to the chat room